src/workers/customTS.worker.ts

```diff
@@ -35,6 +36,9 @@ self.customTSWorkerFactory = tsw => {
 
     async getSemanticDiagnostics(fileName: string): Promise<Diagnostic[]> {
       const diagnostics = await super.getSemanticDiagnostics(fileName);
+
+      diagnostics.push(...generateSequencingDiagnostics(fileName, this._languageService));
```

Comment:
So for the scheduling editor we still get these diagnostic errors if you type a time string (unlikely case, but still potentially a problem). How do we go about only applying these diagnostics to the sequencing editor and command expansion rule editor?

Reply:
I really should implement the whole persistent worker-model context system to get this going, which means among other things I'll probably need an object format to pass through metadata, method overrides to enable, and the command dict.

I'll mess with it this weekend to get a feel for what might work, but we should sync up on Monday about this since I think it might be important to do right for the future.

Reply:
There's also a lot more to architecting the worker side stuff, so this might bleed into a new issue? Or we can just use this here to setup the foundations.

Reply:
Sounds good! Just FYI I'll be on vacation for two weeks starting today. Please tag up with @duranb about UI architecture questions. Certainly create a new issue for architecting this if it needs more time/thought. That is 100% the most important thing about this for future sustainability.